admin/sql/updates/20200326-mbs-7781.sql

```diff
@@ -0,0 +1,84 @@
+\set ON_ERROR_STOP 1
+
+BEGIN;
+
+-- N.B. The artist credit cache should be cleared after running this.
+
+CREATE OR REPLACE FUNCTION merge_duplicate_artist_credits()
+RETURNS void AS $$
+DECLARE
+    dupe_ac RECORD;
+BEGIN
+    FOR dupe_ac IN (
+        -- Select all pairs of dupes, preferring the smaller ID as the
+        -- "original" artist credit.
+        WITH dupe_acs AS (
+            WITH ac_names AS (
+                SELECT ac.id,
+                       array_agg(
+                        row(acn.position,
+                            acn.artist,
+                            acn.name,
+                            acn.join_phrase)
+                        ORDER BY acn.position ASC
+                       ) AS names
+                  FROM artist_credit ac
+                  JOIN artist_credit_name acn
+                    ON acn.artist_credit = ac.id
+                 GROUP BY ac.id
+            )
+            SELECT ac1.id AS orig, ac2.id AS dupe
+              FROM ac_names ac1, ac_names ac2
+             WHERE ac1.id < ac2.id
+               AND ac1.names = ac2.names
+        )
+        -- Remove dupes of other dupes.
+        SELECT *
+          FROM dupe_acs dac1
+         WHERE NOT EXISTS (
+           SELECT 1 FROM dupe_acs dac2
+            WHERE dac2.dupe = dac1.orig
+         )
+         ORDER BY dac1.orig ASC
```

Comment:
So the idea here is that all dupes will be paired with every single dupe with a smaller ID, and this removes those to leave only the pairs between all dupes and the smallest ID of them all?

Reply:
Sounds about right. But there might be a more straightforward way to do that, it was pretty late when I wrote this. :)

Reply:
Ok yeah, a much simpler way is just to group `ac_names` by `names` and stuff all the dupe IDs into one array (with the smallest ID being the first in the array). Did that and also separated out the main query into a `duplicate_artist_credits` view, in case we want to see/inspect if duplicate artist credits exist without merging them.